### PR TITLE
Feature/sumv2

### PR DESCRIPTION
**Pull request by swmal**
New implementation of the SUM (see class SumV2.cs) function that handles boolean values correctly. Also some performance improvements compared to previous version (Sum.cs)

### CODE REVIEW

File: src/EPPlusTest/FormulaParsing/Excel/Functions/Math/SumTests.cs [hunk 1]
```diff
@@ -44,5 +43,90 @@ public void ShouldTreatSingleBooleanValuesAsNumbers()
...
+            }
+        }
+
+        [TestMethod]
+        public void ShouldCountNumericStringViaArgument()
+        {
+            using (var package = new ExcelPackage())
+            {
+                var sheet = package.Workbook.Worksheets.Add("test");
+                sheet.Cells["A1"].Value = "1";
```
[the remark this answers is not in the document]
Comment (OssianEPPlus) [reply src/EPPlusTest/FormulaParsing/Excel/Functions/Math/SumTests.cs:81]: the `sheet.Cells["A1"].Value = "1";` line I mean

File: src/EPPlus/FormulaParsing/Excel/Functions/BuiltInFunctions.cs [hunk 2]
```diff
@@ -112,7 +112,7 @@ public BuiltInFunctions()
             Functions["roman"] = new Roman();
             Functions["sin"] = new Sin();
             Functions["sinh"] = new Sinh();
-            Functions["sum"] = new Sum();
+            Functions["sum"] = new SumV2();
```
[the remark this answers is not in the document]
Comment (swmal) [reply src/EPPlus/FormulaParsing/Excel/Functions/BuiltInFunctions.cs:115]: We are still using the old Sum function inside the Subtotal function. This version does a lot of more checks for errors, if cells are hidden, etc and we improve performance by separating this into two different classes. We could potentially rename the old sum to "SumSubtotal" and this new version to just "Sum" to make things clearer.

File: src/EPPlusTest/FormulaParsing/Excel/Functions/Math/SumTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (swmal) [reply src/EPPlusTest/FormulaParsing/Excel/Functions/Math/SumTests.cs:81]: Yes, I have probably changed the test without removing the unnecessary line...

File: src/EPPlusTest/FormulaParsing/Excel/Functions/Math/SumTests.cs [hunk 3]
```diff
@@ -44,5 +43,90 @@ public void ShouldTreatSingleBooleanValuesAsNumbers()
...
+            {
+                var sheet = package.Workbook.Worksheets.Add("test");
+                sheet.Cells["A1"].Value = 2;
+                sheet.Cells["A2"].Formula = "1/0";
+                sheet.Cells["A4"].Formula = "SUM(A1:A2)";
+                sheet.Calculate();
+                var a4val = sheet.Cells["A4"].Value;
+                Assert.AreEqual(ExcelErrorValue.Create(eErrorType.Div0), a4val);
+            }
+        }
```
[the remark this answers is not in the document]
Comment (OssianEPPlus) [reply src/EPPlusTest/FormulaParsing/Excel/Functions/Math/SumTests.cs:130]: added it

File: src/EPPlus/FormulaParsing/Excel/Functions/BuiltInFunctions.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (OssianEPPlus) [reply src/EPPlus/FormulaParsing/Excel/Functions/BuiltInFunctions.cs:115]: Think that might be better

File: src/EPPlusTest/FormulaParsing/Excel/Functions/Math/SumTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (OssianEPPlus) [reply src/EPPlusTest/FormulaParsing/Excel/Functions/Math/SumTests.cs:81]: Makes sense. Remove it then?

**Review by OssianEPPlus — APPROVED** [review]

**Pull request merged**